[Bug]: 2560x1600x144 分辨率设置失败 (ERROR_GEN_FAILURE) - Windows CCD 拓扑数据库兼容问题

### 🏷️ 问题类型

📺 视频串流问题

### 📝 问题描述

环境信息

Sunshine 版本: v2026.324.103456.杂鱼
操作系统: Windows 11 25H2
GPU: NVIDIA GeForce RTX 3080 Ti
物理显示器: 5120x2160 带鱼屏 (21:9)
客户端设备: 联想 Y700 2023 (2560x1600, 16:10)
客户端软件: 月光·阿西西

问题描述
串流时 Zako VDD 虚拟显示器创建成功，但设置 2560x1600x144 分辨率时报 ERROR_GEN_FAILURE 失败，导致串流回退到捕获物理显示器。
改用 2560x1440x144 则正常工作，主屏正常熄灭，串流正常。
关键日志
[INFO] 成功配置VDD设备: {1f153e55-7f6a-55a0-8436-f8d7eb893693}
[INFO] 应用vdd_prep: 关闭物理显示器
[WARNING] failed to change topology using the topology from Windows DB! Asking Windows to create the topology.
[INFO] Changing display modes to: {1f153e55-7f6a-55a0-8436-f8d7eb893693} -> 2560x1600x144
[INFO] Failed to change display modes using Windows recommended modes, trying to set modes more strictly!
[ERROR] failed to set display mode! code: ERROR_GEN_FAILURE
[ERROR] Failed to set new display modes (resolution + refresh rate)
[WARNING] Client-specified display [ZakoHDR] (resolved: ZakoHDR) not found, using default display
复现步骤

物理显示器分辨率为 5120x2160
Moonlight 客户端设置分辨率为 2560x1600，帧率 144
发起串流连接
Zako VDD 创建成功但设置分辨率失败

已尝试的解决方案

完整卸载重装 Sunshine 基地版
在 vdd_settings.xml 中手动添加 2560x1600 分辨率
重新安装 Zako VDD 驱动
重置显示器设置
均无效，改用 2560x1440 可正常工作

猜测原因
Windows CCD 拓扑数据库中没有 Zako HDR + 2560x1600x144 的记录，且物理显示器为 5120x2160 带鱼屏时，Windows 无法为虚拟显示器创建该拓扑记录。

### 🔄 复现步骤

1.物理显示器为 5120x2160 带鱼屏
2.Moonlight 客户端设置分辨率 2560x1600 / 144fps
3.发起串流连接

### 🎯 受影响的游戏/应用

_No response_

### 🔁 可复现性

None

### 💻 主机操作系统

Windows 11

### 📌 Windows 具体版本

windows 11 25H2

### ☀️ Sunshine 版本

v2026.324.103456

### 🎨 显卡厂商

NVIDIA

### 🎨 显卡型号及驱动

RTX 3080 TI/591.86

### 🔧 视频编码器

None

### 📱 客户端设备

月光·阿西西 12.1-250721-axi

### ⚙️ 是否修改过配置？

None

### 📋 已修改的配置

_No response_

### 📸 截图/视频

_No response_

### 📜 日志输出

```shell
[2026-04-13 01:04:57.181]: Info: Launch request from IP: 192.168.0.235, Port: 55658
[2026-04-13 01:04:57.184]: Info: Available display devices: 
DEVICE ID: {bdb3a875-2ec4-50c8-af8c-55948addfa89}
DISPLAY NAME: \\.\DISPLAY1
FRIENDLY NAME: DP2
DEVICE STATE: PRIMARY
HDR STATE: UNKNOWN
[2026-04-13 01:04:57.185]: Warning: Device ZakoHDR not found in the list of available devices!
[2026-04-13 01:04:57.189]: Info: Available display devices: 
DEVICE ID: {bdb3a875-2ec4-50c8-af8c-55948addfa89}
DISPLAY NAME: \\.\DISPLAY1
FRIENDLY NAME: DP2
DEVICE STATE: PRIMARY
HDR STATE: UNKNOWN
[2026-04-13 01:04:57.189]: Warning: Device ZakoHDR not found in the list of available devices!
[2026-04-13 01:04:57.194]: Info: 已保存pre-VDD设备列表: 
DEVICE ID: {bdb3a875-2ec4-50c8-af8c-55948addfa89}
DISPLAY NAME: \\.\DISPLAY1
FRIENDLY NAME: DP2
DEVICE STATE: PRIMARY
HDR STATE: UNKNOWN
[2026-04-13 01:04:57.195]: Info: 创建虚拟显示器...
[2026-04-13 01:04:57.195]: Info: 创建虚拟显示器，客户端标识符: 6E809B88-0869-B60C-5106-7FC477755830, GUID: {290c7fbc-9b2a-58c6-aff5-5dd1560c85fc}, HDR亮度范围: [1000, 0.001, 1000]
[2026-04-13 01:04:57.925]: Info: 创建虚拟显示器完成，响应:  [return=1]
[2026-04-13 01:04:58.129]: Info: 成功配置VDD设备: {1f153e55-7f6a-55a0-8436-f8d7eb893693}
[2026-04-13 01:04:58.130]: Info: 使用pre-VDD设备列表: 1个物理显示器, 原主屏: {bdb3a875-2ec4-50c8-af8c-55948addfa89}
[2026-04-13 01:04:58.131]: Info: 应用vdd_prep: 关闭物理显示器
[2026-04-13 01:04:58.131]: Info: 成功应用vdd_prep设置
[2026-04-13 01:04:58.132]: Info: 已应用VDD屏幕布局设置
[2026-04-13 01:04:58.336]: Info: 虚拟显示器HDR状态已是目标状态
[2026-04-13 01:04:58.336]: Info: Session lock check - Locked: false, SessionFlags: 1 (LOCK=0, UNLOCK=1)
[2026-04-13 01:04:58.366]: Info: Applying configuration to the display device.
[2026-04-13 01:04:58.385]: Info: VDD mode: topology controlled by vdd_prep in prepare_vdd, only getting current topology metadata
[2026-04-13 01:04:58.391]: Info: Available display devices: 
DEVICE ID: {1f153e55-7f6a-55a0-8436-f8d7eb893693}
DISPLAY NAME: \\.\DISPLAY8
FRIENDLY NAME: Zako HDR
DEVICE STATE: PRIMARY
HDR STATE: DISABLED
-----------------------
DEVICE ID: {bdb3a875-2ec4-50c8-af8c-55948addfa89}
DISPLAY NAME: NOT AVAILABLE
FRIENDLY NAME: DP2
DEVICE STATE: INACTIVE
HDR STATE: UNKNOWN
[2026-04-13 01:04:58.393]: Info: Changing display modes to: 
{1f153e55-7f6a-55a0-8436-f8d7eb893693} -> 2560x1600x144
[2026-04-13 01:04:58.461]: Info: Failed to change display modes using Windows recommended modes, trying to set modes more strictly!
[2026-04-13 01:04:58.478]: Error: [code: ERROR_GEN_FAILURE, message: ����ϵͳ�ϵ��豸û�з������á�

] failed to set display mode!
[2026-04-13 01:04:58.480]: Error: Failed to set display mode(-s) completely!
[2026-04-13 01:04:58.481]: Error: Failed to configure display:
Failed to set new display modes (resolution + refresh rate)
[2026-04-13 01:04:58.484]: Info: apply_config 未执行（无persistent_data），销毁VDD并跳过拓扑恢复
[2026-04-13 01:04:59.678]: Info: 销毁虚拟显示器完成，响应: 
[2026-04-13 01:05:04.060]: Info: apply_config 从未执行成功，跳过拓扑恢复
[2026-04-13 01:05:04.066]: Info: Encoder reenumeration is required
[2026-04-13 01:05:04.067]: Info: Testing for available encoders - Errors during this phase can be ignored (测试可用编码器 - 此阶段的错误可以忽略)
[2026-04-13 01:05:04.067]: Info: Trying encoder [nvenc]
[2026-04-13 01:05:04.069]: Info: Running in service mode, using DDX capture (WGC not available in services)
[2026-04-13 01:05:04.323]: Info: Device Description : NVIDIA GeForce RTX 3080 Ti, Vendor ID: 0x000010DE, Device ID: 0x00002208, Video Mem: 12084 MiB, Feature Level: 0x0000B100, Capture: 5120x2160, Offset: 0x0
[2026-04-13 01:05:04.324]: Info: HAGS: enabled, GPU priority: realtime
[2026-04-13 01:05:04.324]: Info: HDR: DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709, Bits: 10, Luminance: 0.5/270/270 nits
[2026-04-13 01:05:04.325]: Info: Capture gamma: sRGB (G22)
[2026-04-13 01:05:04.326]: Info: Desktop resolution [5120x2160], Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM], Display refresh rate [60Hz], Requested frame rate [60fps]
[2026-04-13 01:05:04.327]: Info: Creating encoder [h264_nvenc], Color coding: SDR (Rec. 601), Color depth: 8-bit, Color range: JPEG
[2026-04-13 01:05:04.379]: Info: NvEnc: created encoder v1300 H.264 P1 async cbr two-pass rfi
[2026-04-13 01:05:04.421]: Info: Creating encoder [hevc_nvenc], Color coding: SDR (Rec. 601), Color depth: 8-bit, Color range: JPEG
[2026-04-13 01:05:04.469]: Info: NvEnc: created encoder v1300 HEVC P1 async cbr two-pass rfi
[2026-04-13 01:05:04.504]: Info: Creating encoder [av1_nvenc], Color coding: SDR (Rec. 601), Color depth: 8-bit, Color range: JPEG
[2026-04-13 01:05:04.531]: Error: NvEnc: encoding format is not supported by the gpu, NVENCAPI_VERSION: 13
[2026-04-13 01:05:04.532]: Error: NvEnc: NvEncUnregisterAsyncEvent() failed: NV_ENC_ERR_DEVICE_NOT_EXIST
[2026-04-13 01:05:04.543]: Info: Creating encoder [h264_nvenc], Color coding: SDR (Rec. 601), Color depth: 8-bit, Color range: MPEG
[2026-04-13 01:05:04.595]: Info: NvEnc: created encoder v1300 H.264 P1 async cbr two-pass rfi
[2026-04-13 01:05:04.654]: Info: Running in service mode, using DDX capture (WGC not available in services)
[2026-04-13 01:05:04.901]: Info: Device Description : NVIDIA GeForce RTX 3080 Ti, Vendor ID: 0x000010DE, Device ID: 0x00002208, Video Mem: 12084 MiB, Feature Level: 0x0000B100, Capture: 5120x2160, Offset: 0x0
[2026-04-13 01:05:04.902]: Info: HAGS: enabled, GPU priority: realtime
[2026-04-13 01:05:04.902]: Info: HDR: DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709, Bits: 10, Luminance: 0.5/270/270 nits
[2026-04-13 01:05:04.903]: Info: Capture gamma: sRGB (G22)
[2026-04-13 01:05:04.904]: Info: Desktop resolution [5120x2160], Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM], Display refresh rate [60Hz], Requested frame rate [60fps]
[2026-04-13 01:05:04.904]: Info: Creating encoder [hevc_nvenc], Color coding: SDR (Rec. 601), Color depth: 10-bit, Color range: MPEG
[2026-04-13 01:05:05.031]: Info: NvEnc: created encoder v1300 HEVC P1 async yuv444 10-bit cbr two-pass rfi
[2026-04-13 01:05:05.095]: Info: Ignore any errors, Encoder testing completed (忽略任何错误，编码器测试完成)
[2026-04-13 01:05:05.095]: Info: Found H.264 encoder: h264_nvenc [nvenc]
[2026-04-13 01:05:05.096]: Info: Found HEVC encoder: hevc_nvenc [nvenc]
[2026-04-13 01:05:05.096]: Info: Executing [Desktop]
[2026-04-13 01:05:05.132]: Info: RTSP DESCRIBE encryption flags: supported=0xd, requested=0x1 (CONTROL_V2=1, VIDEO=0, AUDIO=0, MIC=0)
[2026-04-13 01:05:05.194]: Info: Client requested stream resolution (clientViewport): 2560x1600
[2026-04-13 01:05:05.195]: Info: Client framerate: 144 fps (144/1)
[2026-04-13 01:05:05.195]: Info: New streaming session started [active sessions: 1]
[2026-04-13 01:05:05.196]: Info: Microphone socket started on port 48001
[2026-04-13 01:05:05.197]: Info: Client unknown: Microphone socket closed (session doesn't require it)
[2026-04-13 01:05:05.414]: Info: nvprefs: No need to modify application profile settings
[2026-04-13 01:05:05.415]: Info: nvprefs: No need to modify global profile settings
[2026-04-13 01:05:05.477]: Info: CLIENT CONNECTED
[2026-04-13 01:05:05.661]: Info: Virtual audio device will use 16-bit to match default device
[2026-04-13 01:05:05.675]: Info: Changed virtual audio sink format to [S16 48000 2.0]
[2026-04-13 01:05:05.820]: Warning: Client-specified display [ZakoHDR] (resolved: ZakoHDR) not found, using default display
[2026-04-13 01:05:05.821]: Info: Running in service mode, using DDX capture (WGC not available in services)
[2026-04-13 01:05:05.821]: Info: Audio mixer format is 32-bit, 48000 Hz, no resampling needed
[2026-04-13 01:05:05.826]: Info: Audio capture format is [F32 48000 2.0]
[2026-04-13 01:05:05.826]: Info: Audio capture initialized successfully, entering sampling loop
[2026-04-13 01:05:05.827]: Info: Opus DRED enabled: 100ms redundancy
[2026-04-13 01:05:05.828]: Info: Opus initialized: 48 kHz, 2 channels, 512 kbps (total), LOWDELAY
[2026-04-13 01:05:05.832]: Info: Audio encryption status: encryptionFlagsEnabled=0x5, SS_ENC_AUDIO (0x04) check: enabled, will ENCRYPT audio data
[2026-04-13 01:05:05.833]: Info: Audio cipher status: initialized=yes, key_size=16
[2026-04-13 01:05:05.834]: Info: Audio packet encryption: plaintext_size=320, encrypted_size=336, encryption SUCCESS (data is encrypted)
[2026-04-13 01:05:06.046]: Info: Device Description : NVIDIA GeForce RTX 3080 Ti, Vendor ID: 0x000010DE, Device ID: 0x00002208, Video Mem: 12084 MiB, Feature Level: 0x0000B100, Capture: 5120x2160, Offset: 0x0
[2026-04-13 01:05:06.046]: Info: HAGS: enabled, GPU priority: realtime
[2026-04-13 01:05:06.047]: Info: Fractional framerate: 144/1 (144 fps)
[2026-04-13 01:05:06.048]: Info: HDR: DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709, Bits: 10, Luminance: 0.5/270/270 nits
[2026-04-13 01:05:06.048]: Info: Capture gamma: sRGB (G22)
[2026-04-13 01:05:06.049]: Info: Desktop resolution [5120x2160], Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM], Display refresh rate [60Hz], Requested frame rate [144fps]
[2026-04-13 01:05:06.050]: Info: Initial display: 5120x2160, encoding: 2560x1600, scale: 0.5x0.740741
[2026-04-13 01:05:06.050]: Info: Creating encoder [hevc_nvenc], Color coding: SDR (Rec. 709), Color depth: 8-bit, Color range: MPEG
[2026-04-13 01:05:06.057]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2026-04-13 01:05:06.096]: Info: NvEnc: created encoder v1300 HEVC P1 async cbr two-pass rfi
[2026-04-13 01:05:06.097]: Info: Downscaling from 5120x2160 to 2560x1600 using quality: balanced (bicubic)
[2026-04-13 01:05:06.098]: Info: Minimum frame time set to 13.8889ms, based on client-requested target framerate 144.
[2026-04-13 01:05:07.403]: Info: Creating virtual touch input device
[2026-04-13 01:05:11.126]: Info: CLIENT DISCONNECTED
[2026-04-13 01:05:11.129]: Info: Audio capture sampling loop ended (shutdown requested)
[2026-04-13 01:05:11.131]: Info: Starting async encoder teardown
[2026-04-13 01:05:11.158]: Info: Async encoder teardown complete
[2026-04-13 01:05:11.391]: Info: Cancel request from IP: 192.168.0.235, Port: 43032
[2026-04-13 01:05:17.222]: Info: Launch request from IP: 192.168.0.235, Port: 43100
[2026-04-13 01:05:17.226]: Info: Available display devices: 
DEVICE ID: {bdb3a875-2ec4-50c8-af8c-55948addfa89}
DISPLAY NAME: \\.\DISPLAY1
FRIENDLY NAME: DP2
DEVICE STATE: PRIMARY
HDR STATE: UNKNOWN
[2026-04-13 01:05:17.227]: Warning: Device ZakoHDR not found in the list of available devices!
[2026-04-13 01:05:17.230]: Info: Available display devices: 
DEVICE ID: {bdb3a875-2ec4-50c8-af8c-55948addfa89}
DISPLAY NAME: \\.\DISPLAY1
FRIENDLY NAME: DP2
DEVICE STATE: PRIMARY
HDR STATE: UNKNOWN
[2026-04-13 01:05:17.230]: Warning: Device ZakoHDR not found in the list of available devices!
[2026-04-13 01:05:17.236]: Info: 已保存pre-VDD设备列表: 
DEVICE ID: {bdb3a875-2ec4-50c8-af8c-55948addfa89}
DISPLAY NAME: \\.\DISPLAY1
FRIENDLY NAME: DP2
DEVICE STATE: PRIMARY
HDR STATE: UNKNOWN
[2026-04-13 01:05:17.236]: Info: 创建虚拟显示器...
[2026-04-13 01:05:17.237]: Info: 创建虚拟显示器，客户端标识符: 6E809B88-0869-B60C-5106-7FC477755830, GUID: {290c7fbc-9b2a-58c6-aff5-5dd1560c85fc}, HDR亮度范围: [1000, 0.001, 1000]
[2026-04-13 01:05:17.952]: Info: 创建虚拟显示器完成，响应:  [return=1]
[2026-04-13 01:05:18.157]: Info: 成功配置VDD设备: {1f153e55-7f6a-55a0-8436-f8d7eb893693}
[2026-04-13 01:05:18.157]: Info: 使用pre-VDD设备列表: 1个物理显示器, 原主屏: {bdb3a875-2ec4-50c8-af8c-55948addfa89}
[2026-04-13 01:05:18.158]: Info: 应用vdd_prep: 关闭物理显示器
[2026-04-13 01:05:18.159]: Info: 成功应用vdd_prep设置
[2026-04-13 01:05:18.159]: Info: 已应用VDD屏幕布局设置
[2026-04-13 01:05:18.365]: Info: 虚拟显示器HDR状态已是目标状态
[2026-04-13 01:05:18.366]: Info: Session lock check - Locked: false, SessionFlags: 1 (LOCK=0, UNLOCK=1)
[2026-04-13 01:05:18.397]: Info: Applying configuration to the display device.
[2026-04-13 01:05:18.398]: Info: VDD mode: topology controlled by vdd_prep in prepare_vdd, only getting current topology metadata
[2026-04-13 01:05:18.402]: Info: Available display devices: 
DEVICE ID: {1f153e55-7f6a-55a0-8436-f8d7eb893693}
DISPLAY NAME: \\.\DISPLAY8
FRIENDLY NAME: Zako HDR
DEVICE STATE: PRIMARY
HDR STATE: DISABLED
-----------------------
DEVICE ID: {bdb3a875-2ec4-50c8-af8c-55948addfa89}
DISPLAY NAME: NOT AVAILABLE
FRIENDLY NAME: DP2
DEVICE STATE: INACTIVE
HDR STATE: UNKNOWN
[2026-04-13 01:05:18.404]: Info: Changing display modes to: 
{1f153e55-7f6a-55a0-8436-f8d7eb893693} -> 2560x1440x144
[2026-04-13 01:05:18.408]: Info: 检测到HDR切换操作，等待其他显示器操作稳定...
[2026-04-13 01:05:18.411]: Info: Changing hdr states to: 
{1f153e55-7f6a-55a0-8436-f8d7eb893693} -> DISABLED
[2026-04-13 01:05:18.413]: Info: Display device configuration applied.
[2026-04-13 01:05:18.419]: Info: Encoder reenumeration is required
[2026-04-13 01:05:18.420]: Info: Testing for available encoders - Errors during this phase can be ignored (测试可用编码器 - 此阶段的错误可以忽略)
[2026-04-13 01:05:18.420]: Info: Trying encoder [nvenc]
[2026-04-13 01:05:18.421]: Info: Running in service mode, using DDX capture (WGC not available in services)
[2026-04-13 01:05:18.706]: Info: Device Description : NVIDIA GeForce RTX 3080 Ti, Vendor ID: 0x000010DE, Device ID: 0x00002208, Video Mem: 12084 MiB, Feature Level: 0x0000B100, Capture: 2560x1440, Offset: 0x0
[2026-04-13 01:05:18.706]: Info: HAGS: enabled, GPU priority: realtime
[2026-04-13 01:05:18.707]: Info: HDR: DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709, Bits: 8, Luminance: 0/1015.27/1015.27 nits
[2026-04-13 01:05:18.708]: Info: Capture gamma: sRGB (G22)
[2026-04-13 01:05:18.726]: Info: Desktop resolution [2560x1440], Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM], Display refresh rate [144Hz], Requested frame rate [60fps]
[2026-04-13 01:05:18.727]: Info: Creating encoder [h264_nvenc], Color coding: SDR (Rec. 601), Color depth: 8-bit, Color range: JPEG
[2026-04-13 01:05:18.781]: Info: NvEnc: created encoder v1300 H.264 P1 async cbr two-pass rfi
[2026-04-13 01:05:19.811]: Info: Creating encoder [hevc_nvenc], Color coding: SDR (Rec. 601), Color depth: 8-bit, Color range: JPEG
[2026-04-13 01:05:19.874]: Info: NvEnc: created encoder v1300 HEVC P1 async cbr two-pass rfi
[2026-04-13 01:05:19.924]: Info: Creating encoder [av1_nvenc], Color coding: SDR (Rec. 601), Color depth: 8-bit, Color range: JPEG
[2026-04-13 01:05:19.978]: Error: NvEnc: encoding format is not supported by the gpu, NVENCAPI_VERSION: 13
[2026-04-13 01:05:19.979]: Error: NvEnc: NvEncUnregisterAsyncEvent() failed: NV_ENC_ERR_DEVICE_NOT_EXIST
[2026-04-13 01:05:20.007]: Info: Creating encoder [h264_nvenc], Color coding: SDR (Rec. 601), Color depth: 8-bit, Color range: MPEG
[2026-04-13 01:05:20.066]: Info: NvEnc: created encoder v1300 H.264 P1 async cbr two-pass rfi
[2026-04-13 01:05:20.143]: Info: Running in service mode, using DDX capture (WGC not available in services)
[2026-04-13 01:05:20.415]: Info: Device Description : NVIDIA GeForce RTX 3080 Ti, Vendor ID: 0x000010DE, Device ID: 0x00002208, Video Mem: 12084 MiB, Feature Level: 0x0000B100, Capture: 2560x1440, Offset: 0x0
[2026-04-13 01:05:20.416]: Info: HAGS: enabled, GPU priority: realtime
[2026-04-13 01:05:20.417]: Info: HDR: DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709, Bits: 8, Luminance: 0/1015.27/1015.27 nits
[2026-04-13 01:05:20.418]: Info: Capture gamma: sRGB (G22)
[2026-04-13 01:05:20.434]: Info: Desktop resolution [2560x1440], Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM], Display refresh rate [144Hz], Requested frame rate [60fps]
[2026-04-13 01:05:20.435]: Info: Creating encoder [hevc_nvenc], Color coding: SDR (Rec. 601), Color depth: 10-bit, Color range: MPEG
[2026-04-13 01:05:20.584]: Info: NvEnc: created encoder v1300 HEVC P1 async yuv444 10-bit cbr two-pass rfi
[2026-04-13 01:05:20.670]: Info: Ignore any errors, Encoder testing completed (忽略任何错误，编码器测试完成)
[2026-04-13 01:05:20.671]: Info: Found H.264 encoder: h264_nvenc [nvenc]
[2026-04-13 01:05:20.671]: Info: Found HEVC encoder: hevc_nvenc [nvenc]
[2026-04-13 01:05:20.672]: Info: Executing [Desktop]
[2026-04-13 01:05:20.713]: Info: RTSP DESCRIBE encryption flags: supported=0xd, requested=0x1 (CONTROL_V2=1, VIDEO=0, AUDIO=0, MIC=0)
[2026-04-13 01:05:20.760]: Info: Client requested stream resolution (clientViewport): 2560x1440
[2026-04-13 01:05:20.761]: Info: Client framerate: 144 fps (144/1)
[2026-04-13 01:05:20.761]: Info: New streaming session started [active sessions: 1]
[2026-04-13 01:05:20.762]: Info: Microphone socket started on port 48001
[2026-04-13 01:05:20.763]: Info: Client unknown: Microphone socket closed (session doesn't require it)
[2026-04-13 01:05:20.984]: Info: nvprefs: No need to modify application profile settings
[2026-04-13 01:05:20.984]: Info: nvprefs: No need to modify global profile settings
[2026-04-13 01:05:21.202]: Info: CLIENT CONNECTED
[2026-04-13 01:05:21.249]: Info: Audio mixer format is 32-bit, 48000 Hz, no resampling needed
[2026-04-13 01:05:21.256]: Info: Audio capture format is [F32 48000 2.0]
[2026-04-13 01:05:21.265]: Info: Audio capture initialized successfully, entering sampling loop
[2026-04-13 01:05:21.265]: Info: Opus DRED enabled: 100ms redundancy
[2026-04-13 01:05:21.266]: Info: Opus initialized: 48 kHz, 2 channels, 512 kbps (total), LOWDELAY
[2026-04-13 01:05:21.267]: Info: Audio encryption status: encryptionFlagsEnabled=0x5, SS_ENC_AUDIO (0x04) check: enabled, will ENCRYPT audio data
[2026-04-13 01:05:21.267]: Info: Audio cipher status: initialized=yes, key_size=16
[2026-04-13 01:05:21.268]: Info: Audio packet encryption: plaintext_size=320, encrypted_size=336, encryption SUCCESS (data is encrypted)
[2026-04-13 01:05:21.582]: Info: Using client-specified display: \\.\DISPLAY8
[2026-04-13 01:05:21.583]: Info: Running in service mode, using DDX capture (WGC not available in services)
[2026-04-13 01:05:21.861]: Info: Device Description : NVIDIA GeForce RTX 3080 Ti, Vendor ID: 0x000010DE, Device ID: 0x00002208, Video Mem: 12084 MiB, Feature Level: 0x0000B100, Capture: 2560x1440, Offset: 0x0
[2026-04-13 01:05:21.862]: Info: HAGS: enabled, GPU priority: realtime
[2026-04-13 01:05:21.863]: Info: Fractional framerate: 144/1 (144 fps)
[2026-04-13 01:05:21.863]: Info: HDR: DXGI_COLOR_SPACE_RGB_FULL_G22_NONE_P709, Bits: 8, Luminance: 0/1015.27/1015.27 nits
[2026-04-13 01:05:21.864]: Info: Capture gamma: sRGB (G22)
[2026-04-13 01:05:21.911]: Info: Desktop resolution [2560x1440], Desktop format [DXGI_FORMAT_B8G8R8A8_UNORM], Display refresh rate [144Hz], Requested frame rate [144fps]
[2026-04-13 01:05:21.912]: Info: Initial display: 2560x1440, encoding: 2560x1440, scale: 1x1
[2026-04-13 01:05:21.912]: Info: Creating encoder [hevc_nvenc], Color coding: SDR (Rec. 709), Color depth: 8-bit, Color range: MPEG
[2026-04-13 01:05:21.944]: Info: Capture format [DXGI_FORMAT_B8G8R8A8_UNORM]
[2026-04-13 01:05:21.990]: Info: NvEnc: created encoder v1300 HEVC P1 async cbr two-pass rfi
[2026-04-13 01:05:21.991]: Info: Minimum frame time set to 13.8889ms, based on client-requested target framerate 144.
[2026-04-13 01:05:24.056]: Info: Creating virtual touch input device
[2026-04-13 01:05:27.047]: Info: CLIENT DISCONNECTED
[2026-04-13 01:05:27.049]: Info: Audio capture sampling loop ended (shutdown requested)
[2026-04-13 01:05:27.051]: Info: Starting async encoder teardown
[2026-04-13 01:05:27.079]: Info: Async encoder teardown complete
[2026-04-13 01:05:27.249]: Info: Cancel request from IP: 192.168.0.235, Port: 35194
[2026-04-13 01:05:27.403]: Info: 非常驻模式，销毁VDD
[2026-04-13 01:05:28.871]: Info: 销毁虚拟显示器完成，响应: 
[2026-04-13 01:05:30.686]: Info: Session lock check - Locked: false, SessionFlags: 1 (LOCK=0, UNLOCK=1)
[2026-04-13 01:05:30.717]: Info: 正在恢复显示设备设置 (原因: 串流结束)
[2026-04-13 01:05:30.728]: Info: Removed VDD devices from persistent topology (VDD may have been destroyed)
[2026-04-13 01:05:30.729]: Warning: Topology input is empty!
[2026-04-13 01:05:30.730]: Warning: Topology input is empty!
[2026-04-13 01:05:30.730]: Info: Changing display topology back to: []
[2026-04-13 01:05:30.731]: Warning: Initial topology invalid (VDD may have been removed), keeping current topology
[2026-04-13 01:05:30.853]: Info: 显示设备配置已恢复
```

### 💬 其他补充

_No response_

--- a/src/platform/windows/display_device/device_modes.cpp
+++ b/src/platform/windows/display_device/device_modes.cpp
@@ -250,35 +250,41 @@ namespace display_device {
         // This avoids relying on Windows CCD's automatic source-to-target mode matching,
         // which can fail for non-standard resolutions on newly created virtual display paths.
         const UINT32 target_idx { path->targetInfo.targetModeInfoIdx };
-        if (target_idx != DISPLAYCONFIG_PATH_TARGET_MODE_IDX_INVALID && target_idx < display_data->modes.size()) {
-          auto &target_mode_info = display_data->modes[target_idx];
-          if (target_mode_info.infoType == DISPLAYCONFIG_MODE_INFO_TYPE_TARGET) {
-            auto &signal = target_mode_info.targetMode.targetVideoSignalInfo;
-            const UINT32 width = mode.resolution.width;
-            const UINT32 height = mode.resolution.height;
-            const UINT32 vsync_num = mode.refresh_rate.numerator;
-            const UINT32 vsync_den = mode.refresh_rate.denominator > 0 ? mode.refresh_rate.denominator : 1;
-
-            signal.activeSize.cx = width;
-            signal.activeSize.cy = height;
-            signal.totalSize.cx = width;
-            signal.totalSize.cy = height;
-            signal.vSyncFreq.Numerator = vsync_num;
-            signal.vSyncFreq.Denominator = vsync_den;
-            signal.hSyncFreq.Numerator = vsync_num * height;
-            signal.hSyncFreq.Denominator = vsync_den;
-            signal.pixelRate = static_cast<UINT64>(vsync_num) * width * height / vsync_den;
-            signal.scanLineOrdering = DISPLAYCONFIG_SCANLINE_ORDERING_PROGRESSIVE;
-
-            // Clear the desktop image index so Windows reselects it.
-            // A stale DISPLAYCONFIG_MODE_INFO_TYPE_DESKTOP_IMAGE entry (with old size)
-            // can cause SetDisplayConfig to fail with ERROR_GEN_FAILURE when using
-            // SDC_VIRTUAL_MODE_AWARE, even though we are keeping the target index.
-            w_utils::set_desktop_index(*path, boost::none);
-
-            changes_applied = true;
-          }
+        if (target_idx == DISPLAYCONFIG_PATH_TARGET_MODE_IDX_INVALID || target_idx >= display_data->modes.size()) {
+          BOOST_LOG(warning) << "Explicit target mode fallback: no valid target mode entry for " << device_id << ", skipping.";
+          return false;
         }
+
+        auto &target_mode_info = display_data->modes[target_idx];
+        if (target_mode_info.infoType != DISPLAYCONFIG_MODE_INFO_TYPE_TARGET) {
+          BOOST_LOG(warning) << "Explicit target mode fallback: mode entry is not a TARGET type for " << device_id << ", skipping.";
+          return false;
+        }
+
+        auto &signal = target_mode_info.targetMode.targetVideoSignalInfo;
+        const UINT32 width = mode.resolution.width;
+        const UINT32 height = mode.resolution.height;
+        const UINT32 vsync_num = mode.refresh_rate.numerator;
+        const UINT32 vsync_den = mode.refresh_rate.denominator > 0 ? mode.refresh_rate.denominator : 1;
+
+        signal.activeSize.cx = width;
+        signal.activeSize.cy = height;
+        signal.totalSize.cx = width;
+        signal.totalSize.cy = height;
+        signal.vSyncFreq.Numerator = vsync_num;
+        signal.vSyncFreq.Denominator = vsync_den;
+        signal.hSyncFreq.Numerator = vsync_num * height;
+        signal.hSyncFreq.Denominator = vsync_den;
+        signal.pixelRate = static_cast<UINT64>(vsync_num) * width * height / vsync_den;
+        signal.scanLineOrdering = DISPLAYCONFIG_SCANLINE_ORDERING_PROGRESSIVE;
+
+        // Clear the desktop image index so Windows reselects it.
+        // A stale DISPLAYCONFIG_MODE_INFO_TYPE_DESKTOP_IMAGE entry (with old size)
+        // can cause SetDisplayConfig to fail with ERROR_GEN_FAILURE when using
+        // SDC_VIRTUAL_MODE_AWARE, even though we are keeping the target index.
+        w_utils::set_desktop_index(*path, boost::none);
+
+        changes_applied = true;
       }
 
       if (!changes_applied) {

--- a/src/platform/windows/display_device/device_modes.cpp
+++ b/src/platform/windows/display_device/device_modes.cpp
@@ -270,6 +270,12 @@ namespace display_device {
             signal.pixelRate = static_cast<UINT64>(vsync_num) * width * height / vsync_den;
             signal.scanLineOrdering = DISPLAYCONFIG_SCANLINE_ORDERING_PROGRESSIVE;
 
+            // Clear the desktop image index so Windows reselects it.
+            // A stale DISPLAYCONFIG_MODE_INFO_TYPE_DESKTOP_IMAGE entry (with old size)
+            // can cause SetDisplayConfig to fail with ERROR_GEN_FAILURE when using
+            // SDC_VIRTUAL_MODE_AWARE, even though we are keeping the target index.
+            w_utils::set_desktop_index(*path, boost::none);
+
             changes_applied = true;
           }
         }

--- a/src/platform/windows/display_device/device_modes.cpp
+++ b/src/platform/windows/display_device/device_modes.cpp
@@ -273,21 +273,6 @@ namespace display_device {
             changes_applied = true;
           }
         }
-
-        // Also update desktop image info if present
-        const UINT32 desktop_idx { path->targetInfo.desktopModeInfoIdx };
-        if (desktop_idx != DISPLAYCONFIG_PATH_DESKTOP_IMAGE_IDX_INVALID && desktop_idx < display_data->modes.size()) {
-          auto &desktop_mode_info = display_data->modes[desktop_idx];
-          if (desktop_mode_info.infoType == DISPLAYCONFIG_MODE_INFO_TYPE_DESKTOP_IMAGE) {
-            auto &desktop = desktop_mode_info.desktopImageInfo;
-            desktop.PathSourceSize.cx = mode.resolution.width;
-            desktop.PathSourceSize.cy = mode.resolution.height;
-            desktop.DesktopImageRegion.right = mode.resolution.width;
-            desktop.DesktopImageRegion.bottom = mode.resolution.height;
-            desktop.DesktopImageClip.right = mode.resolution.width;
-            desktop.DesktopImageClip.bottom = mode.resolution.height;
-          }
-        }
       }
 
       if (!changes_applied) {

--- a/src/platform/windows/display_device/device_modes.cpp
+++ b/src/platform/windows/display_device/device_modes.cpp
@@ -381,10 +381,7 @@ namespace display_device {
     }
 
     constexpr bool allow_changes { true };
-    if (!do_set_modes(modes, allow_changes)) {
-      // Error already logged
-      return false;
-    }
+    const bool first_attempt_ok { do_set_modes(modes, allow_changes) };
 
     const auto all_modes_match = [&modes](const device_display_mode_map_t &current_modes) {
       for (const auto &[device_id, requested_mode] : modes) {
@@ -403,38 +400,37 @@ namespace display_device {
     };
 
     auto current_modes { get_current_display_modes(device_ids) };
-    if (!current_modes.empty()) {
-      if (all_modes_match(current_modes)) {
+    if (first_attempt_ok && !current_modes.empty() && all_modes_match(current_modes)) {
+      return true;
+    }
+
+    // We have a problem when using SetDisplayConfig with SDC_ALLOW_CHANGES
+    // where it decides to use our new mode merely as a suggestion.
+    //
+    // This is good, since we don't have to be very precise with refresh rate,
+    // but also bad since it can just ignore our specified mode.
+    //
+    // However, it is possible that the user has created a custom display mode
+    // which is not exposed to the via Windows settings app. To allow this
+    // resolution to be selected, we actually need to omit SDC_ALLOW_CHANGES
+    // flag.
+    BOOST_LOG(info) << "Failed to change display modes using Windows recommended modes, trying to set modes more strictly!";
+    if (do_set_modes(modes, !allow_changes)) {
+      current_modes = get_current_display_modes(device_ids);
+      if (!current_modes.empty() && all_modes_match(current_modes)) {
         return true;
       }
+    }
 
-      // We have a problem when using SetDisplayConfig with SDC_ALLOW_CHANGES
-      // where it decides to use our new mode merely as a suggestion.
-      //
-      // This is good, since we don't have to be very precise with refresh rate,
-      // but also bad since it can just ignore our specified mode.
-      //
-      // However, it is possible that the user has created a custom display mode
-      // which is not exposed to the via Windows settings app. To allow this
-      // resolution to be selected, we actually need to omit SDC_ALLOW_CHANGES
-      // flag.
-      BOOST_LOG(info) << "Failed to change display modes using Windows recommended modes, trying to set modes more strictly!";
-      if (do_set_modes(modes, !allow_changes)) {
-        current_modes = get_current_display_modes(device_ids);
-        if (!current_modes.empty() && all_modes_match(current_modes)) {
-          return true;
-        }
-      }
-
-      // Fallback: explicitly set target mode parameters instead of relying on Windows CCD mode matching.
-      // This helps with non-standard resolutions (e.g. 16:10 like 2560x1600) on newly created virtual
-      // display paths where the CCD topology database has no matching mode entries.
-      BOOST_LOG(info) << "CCD mode matching failed, trying to set modes with explicit target mode parameters.";
-      if (do_set_modes_with_explicit_target(modes)) {
-        current_modes = get_current_display_modes(device_ids);
-        if (!current_modes.empty() && all_modes_match(current_modes)) {
-          return true;
-        }
+    // Fallback: explicitly set target mode parameters instead of relying on Windows CCD mode matching.
+    // This helps with non-standard resolutions (e.g. 16:10 like 2560x1600) on newly created virtual
+    // display paths where the CCD topology database has no matching mode entries.
+    // This also handles the case where the first attempt fails with ERROR_GEN_FAILURE.
+    BOOST_LOG(info) << "CCD mode matching failed, trying to set modes with explicit target mode parameters.";
+    if (do_set_modes_with_explicit_target(modes)) {
+      current_modes = get_current_display_modes(device_ids);
+      if (!current_modes.empty() && all_modes_match(current_modes)) {
+        return true;
       }
     }
 

--- a/src/platform/windows/display_device/device_modes.cpp
+++ b/src/platform/windows/display_device/device_modes.cpp
@@ -198,6 +198,112 @@ namespace display_device {
       return true;
     };
 
+    /**
+     * @brief Attempt to set display modes by explicitly specifying both source and target mode parameters.
+     *
+     * This is a fallback for cases where Windows CCD cannot automatically match source modes to target modes
+     * (e.g., non-standard resolutions like 2560x1600 on newly created virtual display paths).
+     * Instead of clearing the target mode index and relying on Windows' mode matching algorithm,
+     * this function directly modifies the existing target mode entry with the desired parameters.
+     *
+     * @param modes The desired display modes to set.
+     * @return True if successfully applied, false otherwise.
+     */
+    bool
+    do_set_modes_with_explicit_target(const device_display_mode_map_t &modes) {
+      auto display_data { w_utils::query_display_config(w_utils::ACTIVE_ONLY_DEVICES) };
+      if (!display_data) {
+        return false;
+      }
+
+      bool changes_applied { false };
+      for (const auto &[device_id, mode] : modes) {
+        auto path { w_utils::get_active_path(device_id, display_data->paths) };
+        if (!path) {
+          BOOST_LOG(error) << "Failed to find device for explicit target mode set: " << device_id << "!";
+          return false;
+        }
+
+        auto source_mode { w_utils::get_source_mode(w_utils::get_source_index(*path, display_data->modes), display_data->modes) };
+        if (!source_mode) {
+          BOOST_LOG(error) << "Active device does not have a source mode: " << device_id << "!";
+          return false;
+        }
+
+        // Update source mode with desired resolution
+        const bool resolution_changed { source_mode->width != mode.resolution.width || source_mode->height != mode.resolution.height };
+        if (resolution_changed) {
+          source_mode->width = mode.resolution.width;
+          source_mode->height = mode.resolution.height;
+          changes_applied = true;
+        }
+
+        // Update path refresh rate
+        const bool refresh_rate_changed { path->targetInfo.refreshRate.Numerator != mode.refresh_rate.numerator ||
+                                          path->targetInfo.refreshRate.Denominator != mode.refresh_rate.denominator };
+        if (refresh_rate_changed) {
+          path->targetInfo.refreshRate = { mode.refresh_rate.numerator, mode.refresh_rate.denominator };
+          changes_applied = true;
+        }
+
+        // Instead of clearing the target index, try to update the target mode entry in-place.
+        // This avoids relying on Windows CCD's automatic source-to-target mode matching,
+        // which can fail for non-standard resolutions on newly created virtual display paths.
+        const UINT32 target_idx { path->targetInfo.targetModeInfoIdx };
+        if (target_idx != DISPLAYCONFIG_PATH_TARGET_MODE_IDX_INVALID && target_idx < display_data->modes.size()) {
+          auto &target_mode_info = display_data->modes[target_idx];
+          if (target_mode_info.infoType == DISPLAYCONFIG_MODE_INFO_TYPE_TARGET) {
+            auto &signal = target_mode_info.targetMode.targetVideoSignalInfo;
+            const UINT32 width = mode.resolution.width;
+            const UINT32 height = mode.resolution.height;
+            const UINT32 vsync_num = mode.refresh_rate.numerator;
+            const UINT32 vsync_den = mode.refresh_rate.denominator > 0 ? mode.refresh_rate.denominator : 1;
+
+            signal.activeSize.cx = width;
+            signal.activeSize.cy = height;
+            signal.totalSize.cx = width;
+            signal.totalSize.cy = height;
+            signal.vSyncFreq.Numerator = vsync_num;
+            signal.vSyncFreq.Denominator = vsync_den;
+            signal.hSyncFreq.Numerator = vsync_num * height;
+            signal.hSyncFreq.Denominator = vsync_den;
+            signal.pixelRate = static_cast<UINT64>(vsync_num) * width * height / vsync_den;
+            signal.scanLineOrdering = DISPLAYCONFIG_SCANLINE_ORDERING_PROGRESSIVE;
+
+            changes_applied = true;
+          }
+        }
+
+        // Also update desktop image info if present
+        const UINT32 desktop_idx { path->targetInfo.desktopModeInfoIdx };
+        if (desktop_idx != DISPLAYCONFIG_PATH_DESKTOP_IMAGE_IDX_INVALID && desktop_idx < display_data->modes.size()) {
+          auto &desktop_mode_info = display_data->modes[desktop_idx];
+          if (desktop_mode_info.infoType == DISPLAYCONFIG_MODE_INFO_TYPE_DESKTOP_IMAGE) {
+            auto &desktop = desktop_mode_info.desktopImageInfo;
+            desktop.PathSourceSize.cx = mode.resolution.width;
+            desktop.PathSourceSize.cy = mode.resolution.height;
+            desktop.DesktopImageRegion.right = mode.resolution.width;
+            desktop.DesktopImageRegion.bottom = mode.resolution.height;
+            desktop.DesktopImageClip.right = mode.resolution.width;
+            desktop.DesktopImageClip.bottom = mode.resolution.height;
+          }
+        }
+      }
+
+      if (!changes_applied) {
+        return false;
+      }
+
+      const UINT32 flags { SDC_APPLY | SDC_USE_SUPPLIED_DISPLAY_CONFIG | SDC_SAVE_TO_DATABASE | SDC_VIRTUAL_MODE_AWARE };
+      const LONG result { SetDisplayConfig(display_data->paths.size(), display_data->paths.data(), display_data->modes.size(), display_data->modes.data(), flags) };
+      if (result != ERROR_SUCCESS) {
+        BOOST_LOG(warning) << w_utils::get_error_string(result) << " failed to set display mode with explicit target!";
+        return false;
+      }
+
+      return true;
+    }
+
   }  // namespace
 
   device_display_mode_map_t
@@ -329,6 +435,17 @@ namespace display_device {
       // flag.
       BOOST_LOG(info) << "Failed to change display modes using Windows recommended modes, trying to set modes more strictly!";
       if (do_set_modes(modes, !allow_changes)) {
+        current_modes = get_current_display_modes(device_ids);
+        if (!current_modes.empty() && all_modes_match(current_modes)) {
+          return true;
+        }
+      }
+
+      // Fallback: explicitly set target mode parameters instead of relying on Windows CCD mode matching.
+      // This helps with non-standard resolutions (e.g. 16:10 like 2560x1600) on newly created virtual
+      // display paths where the CCD topology database has no matching mode entries.
+      BOOST_LOG(info) << "CCD mode matching failed, trying to set modes with explicit target mode parameters.";
+      if (do_set_modes_with_explicit_target(modes)) {
         current_modes = get_current_display_modes(device_ids);
         if (!current_modes.empty() && all_modes_match(current_modes)) {
           return true;

